### PR TITLE
Fix compatibility with node 18

### LINKED ISSUE #49: Fails with node 18
Hey there!

When upgrading our application to node 18 it started throwing this error:

> /node_modules/.pnpm/moment-timezone@0.5.32/node_modules/moment-timezone/index.js:2:15-51 - Error: Module not found: Error: error:0308010C:digit
al envelope routines::unsupported

After some digging I believe the problem is in this plugin as it uses md4:

https://github.com/gilmoreorless/moment-timezone-data-webpack-plugin/blob/main/src/helpers.js#L149

Just trying to run this in node 18 throws the error:

```
console.log(require('crypto').createHash("md4"));
```

Happy to send a PR with the fix!

2 options I can see:

1. change the `md4` => `md5`, downsides to this is it will invalidate the cache when updating as the hash will change
2. conditionally change the hashing algorithm depending on the node version. More code but won't invalidate the cache on update

Let me know what you'd prefer!

### PR DESCRIPTION
**Pull request by mattlewis92**
Opening up for discussion, I don't think it would really be seen as a breaking change for the cache to be invalidated, but open to suggestions!

Fixes #49

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node: [12, 14, 16]
+        node: [12, 14, 16, 18]
```
Comment (gilmoreorless) [inline .github/workflows/tests.yml:24]: 👍

File: test/utils.js [hunk 2]
```diff
@@ -11,11 +11,12 @@ const rGeneratedFile = /[\\/]node_modules[\\/]\.cache[\\/]moment-timezone-data-w
 function buildWebpack(pluginOptions, testOptions = {}) {
   const compilerOptions = {
     mode: 'development',
-    devtool: 'hidden-source-map',
+    devtool: 'eval',
```
[the remark this answers is not in the document]
Comment (mattlewis92) [reply test/utils.js:14]: The `hidden-source-map` option also uses md4 hashing so throws the same error. I couldn't see an option to change this so just changed the source map option to something that didn't use md4

**Comment by gilmoreorless** [comment]
I'll get a release out with this change in the next few days. Just need to check a couple of other compatibility things before I publish it.

File: test/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (gilmoreorless) [reply test/utils.js:14]: Cool, that makes sense. 😄

**Pull request merged**